eortologio@danchris.github.io/helpers.js

```diff
@@ -1,69 +1,70 @@
-const GLib = imports.gi.GLib;
-const Me = imports.misc.extensionUtils.getCurrentExtension();
+import GLib from 'gi://GLib';
+import Gio from 'gi://Gio';
 
-function getNameDays(currentDatetime){
+export function getNameDays(currentDateTime){
 
     let nameDays = [];
-    let date = getCurrentDate(currentDatetime);
-    let [easterDay, easterMonth, easterYear] = calcOrthEaster(currentDatetime.get_year());
-    
-    return nameDays.concat(getRecurringNameDays(date), getRelativeToEasterNameDays(easterDay, easterMonth, easterYear, currentDatetime));
-    
+    let date = getCurrentDate(currentDateTime);
+    let [easterDay, easterMonth, easterYear] = calcOrthEaster(currentDateTime.get_year());
     
+    return nameDays.concat(getRecurringNameDays(date), getRelativeToEasterNameDays(easterDay, easterMonth, easterYear, currentDateTime));
+   
 }
 
-function getRecurringNameDays(date){
+export function getRecurringNameDays(date, subdir){
+    const file = Gio.File.new_for_uri(import.meta.url).get_parent().get_child('recurring_namedays.json'); 
```

Comment:
Here you create a pointer. Please, consider to free the memory properly before return from the function.

Reply:
per andyholmes he said we are not supposed to free the memory properly before the return from the function.

Reply:
We are just assigning a value to a variable, when the scope ends it will be collected (i.e. freed)